### PR TITLE
chore: use decimal places value in vault dashboard data

### PR DESCRIPTION
**Pull request by rabi-siddique**
The PR does the following:
- Creates a `boardAuxes` map containing `decimalPlaces` for a corresponding token. 
- Generates a number with leading zeroes called a `divisor`, and substitutes it wherever division by `1_000_000` occurs.
- Update the existing tests that were failing. 
- Add a test to ensure `boardAuxes` fallback is used in graphs. 

At the moment, I've done this in Vault's route. Once this PR is approved and any concerns are addressed, I'll replicate this for other routes as well.

### CODE REVIEW

File: src/types/vault-types.ts [hunk 1]
```diff
@@ -80,7 +81,12 @@ export type OracleDailyPriceNodesData = { [key: string]: OraclePriceDailiesNode[
...
+    OraclePriceNode &
+    VaultManagerGovernancesNode &
+    OraclePriceDailiesArr & { decimalPlaces: number };
 };
 
-export type OpenVaultsData = Array<VaultsNode & OraclePriceNode & VaultManagerGovernancesNode>;
\ No newline at end of file
+export type OpenVaultsData = Array<
+  VaultsNode & OraclePriceNode & VaultManagerGovernancesNode & { decimalPlaces: number }
+>;
```
Comment (frazarshad) [inline src/types/vault-types.ts:92]: why dont we use `BoardAuxesNode` instead of `{ decimalPlaces: number }`

File: src/pages/Vaults.tsx [hunk 2]
```diff
@@ -117,18 +136,21 @@ function processVaultsData(
       throw new Error(`Node ID does not contain enough segments: ${node.id}`);
     }
     const managerName = idSegments.slice(0, 4).join('.');
-    const {liquidatingCollateralBrand} = node;
+
+    const liquidatingCollateralBrand = node.liquidatingCollateralBrand;
+    const decimalPlaces = (boardAuxes && boardAuxes[liquidatingCollateralBrand]) || 6;
```
Comment (frazarshad) [inline src/pages/Vaults.tsx:141]: keeping a default value of 6 here could potentially obscure any issue on the frontend.
we might have a missing boardAux and due to this safety check we might be using another value instead

something like 0 might be a better decision to use here?

File: src/components/VaultCharts.tsx [hunk 3]
```diff
@@ -3,32 +3,41 @@ import useSWR from 'swr/immutable';
...
     const oracle = (dailyOracles && dailyOracles[dateKey]) || { typeOutAmountLast: 1, typeInAmountLast: 1 };
     const blockTime = dailyTokenMetrics?.blockTimeLast?.slice(0, 10);
     const liquidatingCollateralBrand = dailyTokenMetrics?.liquidatingCollateralBrand;
-    const totalCollateralLast = dailyTokenMetrics?.totalCollateralLast;
-    const totalDebtLast = dailyTokenMetrics?.totalDebtLast / 1_000_000;
+    const decimalPlaces = (boardAuxes && boardAuxes[liquidatingCollateralBrand]) || 6;
+    const divisor = createNumberWithLeadingZeroes(decimalPlaces);
+
+    const totalCollateralLast = Number(dailyTokenMetrics?.totalCollateralLast);
+    const totalDebtLast = Number(dailyTokenMetrics?.totalDebtLast) / divisor;
```
Comment (frazarshad) [inline src/components/VaultCharts.tsx:37]: i dont think the divisor is used in this case

Comment (frazarshad) [reply src/components/VaultCharts.tsx:37]: since this field represents IST, and the divisor is only for other tokens like stATOM, stOSMO etc

**Review by frazarshad — COMMENTED** [review]
@toliaqat can you please verify that all the places that the divisor is used is indeed correct? as far as i am aware it is only to be used in certain places i.e in values that represent a token such as stkATOM, stATOM, stOSMO etc

**Review by frazarshad — COMMENTED** [review]
@toliaqat can you please verify that all the places that the divisor is used is indeed correct? as far as i am aware it is only to be used in certain places i.e in values that represent a token such as stkATOM, stATOM, stOSMO etc

File: src/types/vault-types.ts (hunk 1 above)
Comment (rabi-siddique) [reply src/types/vault-types.ts:92]: `BoardAuxesNode` has `allegedName` property as well. That is not required.
But now that you've mentioned it, I think we should use `Pick<BoardAuxesNode, 'decimalPlaces'>;` for clarity?

File: src/pages/Vaults.tsx (hunk 2 above)
Comment (rabi-siddique) [reply src/pages/Vaults.tsx:141]: Perhaps. But at the moment, we only get `boardAuxes` data for `ATOM`. For now, the fallback of `6` works for tokens whose data is not fetched by `boardAuxes` query.

File: src/components/VaultCharts.tsx (hunk 3 above)
Comment (rabi-siddique) [reply src/components/VaultCharts.tsx:37]: It was a recent change [over here](https://github.com/agoric-labs/agoric-inter-dashboard/commit/79ec8ad517f714ab7da53a024794fdfa95b6c447)

**Comment by rabi-siddique** [comment]
@frazarshad Thanks for a thorough review. Made the changes.

File: tests/PSM.test.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ jest.mock('recharts', () => {
   return {
     ...OriginalModule,
     ResponsiveContainer: ({ children }: { children: typeof React.Children }) => (
-      <OriginalModule.ResponsiveContainer width={800} height={800}>
+      <OriginalModule.ResponsiveContainer width="100%" height={300}>
```
Comment (frazarshad) [inline tests/PSM.test.tsx:25]: lets keep the original widths. while that might result in warnings, its also gives a fixed non-zero width to the component being rendered. 
If we give a percentage instead, it results in the width of the component becoming 0 and the table component inside the page not being rendered. hence the update we are seeing in the snapshot.

File: tests/Reserve.test.tsx [hunk 5]
```diff
@@ -28,7 +28,7 @@ jest.mock('recharts', () => {
   return {
     ...OriginalModule,
     ResponsiveContainer: ({ children }: { children: typeof React.Children }) => (
-      <OriginalModule.ResponsiveContainer width={800} height={800}>
+      <OriginalModule.ResponsiveContainer width="100%" height={300}>
```
Comment (frazarshad) [inline tests/Reserve.test.tsx:31]: same here

File: tests/PSM.test.tsx (hunk 4 above)
Comment (rabi-siddique) [reply tests/PSM.test.tsx:25]: How come? I am seeing many components using widths in `%`.

**Review by frazarshad — APPROVED** [review]

**Pull request merged**